#! /bin/sh /usr/share/dpatch/dpatch-run
## 07_avoid_warnings.dpatch by Roland Rosenfeld <roland@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Some changes to avoid compiler warnings.

@DPATCH@
diff -urNad transfig~/fig2dev/dev/genpstex.c transfig/fig2dev/dev/genpstex.c
--- transfig~/fig2dev/dev/genpstex.c
+++ transfig/fig2dev/dev/genpstex.c
@@ -167,6 +167,10 @@
 	INCLUDE_TEXT
 };
 
+extern void     genpdf_option();
+extern void     genpdf_start();
+extern int      genpdf_end();
+
 struct driver dev_pdftex = {
   	genpdf_option,
 	genpdf_start,
diff -urNad transfig~/transfig/sys.c transfig/transfig/sys.c
--- transfig~/transfig/sys.c
+++ transfig/transfig/sys.c
@@ -15,6 +15,8 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
 #include "transfig.h"
 
 #define MAXSYS 10000
diff -urNad transfig~/transfig/transfig.c transfig/transfig/transfig.c
--- transfig~/transfig/transfig.c
+++ transfig/transfig/transfig.c
@@ -25,6 +25,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "patchlevel.h"
 #include "transfig.h"
 
diff -urNad transfig~/transfig/txfile.c transfig/transfig/txfile.c
--- transfig~/transfig/txfile.c
+++ transfig/transfig/txfile.c
@@ -15,6 +15,7 @@
  */
 
 #include <stdio.h>
+#include <stdlib.h>
 #include "transfig.h"
 
 /*
